pkg/installmanager/installmanager.go

```diff
@@ -619,14 +667,56 @@ func (m *InstallManager) loadClusterDeployment() (*hivev1.ClusterDeployment, err
 	return cd, nil
 }
 
-func gatherLogs(cd *hivev1.ClusterDeployment, m *InstallManager) error {
-	m.log.Info("Gathering logs/tarball")
+func isGatherLogsEnabled() bool {
+	// By default we assume to gather logs, only disable if explicitly told to via HiveConfig.
+	envVarValue := os.Getenv(constants.SkipGatherLogsEnvVar)
+	return envVarValue == "false" || envVarValue == ""
+}
+
+// gatherLogs will attempt to gather logs after a failed install. First we attempt
+// to gather logs from the bootstrap node. If this fails, we may have made it far enough
+// to teardown the bootstrap node, in which case we then attempt to gather with
+// 'oc adm must-gather', which would gather logs from the cluster's API itself.
+// If neither succeeds we do not consider this a fatal error,
+// we're just gathering as much information as we can and then proceeding with cleanup
+// so we can re-try.
+func (m *InstallManager) gatherLogs(cd *hivev1.ClusterDeployment) {
+
+	err := m.gatherBootstrapNodeLogs(cd)
```

Comment:
I would like the two sections of this function to have similar structure. Also, we do not need a warning log unless both methods fail. We can log the error associated with each method as an info log to cover scenarios where bootstrap logging may be failing when it should be succeeding.
```
if err := m.gatherBootstrapNodeLogs(cd); err != nil {
  m.log.Info(...)
  return
} else {
  m.log.WithError(err).Warn(...)
}
if err := m.gatherClusterLogs(cd); err != nil {
  m.log.Info(...)
  return
} else {
  m.log.WithError(err).Warn(...)
}
m.log.Warn("could not fetch logs") // or some such message
```